docs/rst/tutorials/core/deployment/ds_wan_tcp/ds_wan_tcp.rst

```diff
@@ -0,0 +1,280 @@
+
+.. literalinclude:: /resources/tutorials/core/deployment/ds_wan_tcp_tutorial/Dockerfile
+    :language: Dockerfile
+
+Finally, the ``compose.yml`` is where all the containers and their configuration are described:
+
+* ``ros_talker``: the container is included in the created ``talker_net``.
+  The IP address has not been set manually, but the IP assigned by default would be ``10.1.0.2``.
+  The environment variable ``FASTRTPS_DEFAULT_PROFILES_FILE`` is set with the ``node_configuration.xml`` and ``ROS_DISCOVERY_SERVER`` is set with the discovery *server* information ``"TCPv4:[10.1.1.1]:10111"``.
+  The default gateway of this container is redirected to the discovery *server*, which would behave as a router too.
+
+* ``ros_listener``: the container is included in the created ``listener_net``.
+  The IP address has not been set manually, but the IP assigned by default would be ``10.2.0.2``.
+  As the previous container, the environment variable ``FASTRTPS_DEFAULT_PROFILES_FILE`` is set with the ``node_configuration.xml`` and ``ROS_DISCOVERY_SERVER`` is set with the discovery *server* information ``"TCPv4:[10.1.1.1]:10111"``.
+  The default gateway of this container is redirected to the *router* container.
+
+* ``fast_dds_discovery_server``: the container is included in both created ``talker_net`` and ``wan_net``.
+  The IP addresses has been set manually as ``10.1.1.1`` in the ``talker_net``, and ``10.3.1.1`` in the ``wan_net``.
+  The default gateway of this container is redirected to the *router* container.
+  The ``iptables`` has been configured to redirect any traffic from any network and interface.
+
+* ``router``: the container is included in both ``listener_net`` and ``wan_net`` networks.
+  The IP addresses has been manually set as ``10.2.1.1`` in the ``listener_net``, and ``10.3.2.1`` in the ``wan_net``.
+  The default gateway of this container is redirected to the discovery *server* container.
+  The ``iptables`` has been configured to redirect traffic from any network and interface.
+
+
+Please, include the following ``compose.yml`` file in the workspace.
+
+.. literalinclude:: /resources/tutorials/core/deployment/ds_wan_tcp_tutorial/compose.yml
+    :language: yaml
+
+Run the example
+---------------
+
+Make sure that the workspace has been set with all the previous files, and the docker networks and ``iptables`` have been set too.
+
+.. code-block:: bash
+
+    ├ <workspace>
+    ·  ├ compose.yml
+    ·  ├ Dockerfile
+    ·  ├ node_configuration.xml
+    ·  └ server_configuration.xml
+
+As explained in the :ref:`tutorials_deployment_ds_wan_tcp_overview`, the expected behavior is that both ``talker`` and ``listener`` are able to connect to the discovery *server*, discover each other, and send and receive (respectively) the *HelloWorld* example messages over the WAN simulation using TCP.
+
+Run the example:
+
+.. code-block:: bash
+
+    cd <workspace>
+    docker compose -f compose.yml up --build
+
+.. note::
```

Comment:
I guess it would be worth mentioning at somewhere in the tutorial that when deploying under a real scenario, the proper firewall/router configuration rules needs to be added.